docs/source/developers/guide/step_by_step/building.rst

```diff
@@ -35,17 +35,83 @@
 Building Arrow's libraries 🏋🏿‍♀️
 *********************************
 
+If you decide to contribute to Arrow you will meet the topic of
+compiling source code and use of CMake. You may have some
```

Comment:
The mention of CMake applies mostly to Arrow C++ and C++-derived bindings such as Python or R (not Java or Go, for example). Perhaps you could start this document by mentioning this context?

Reply:
Oh, didn't even know that. Will have to learn about other libraries to be able to explain the difference. Thanks for pointing it out!

Reply:
Well, it's nicely explained in https://arrow.apache.org/overview/ 🤦‍♀️